validator/client/src/test/java/tech/pegasys/teku/validator/client/restapi/apis/SetGraffitiTest.java

```diff
@@ -0,0 +1,143 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_FORBIDDEN;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_INTERNAL_SERVER_ERROR;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NOT_FOUND;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NO_CONTENT;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_UNAUTHORIZED;
+import static tech.pegasys.teku.infrastructure.restapi.MetadataTestUtil.verifyMetadataEmptyResponse;
+import static tech.pegasys.teku.infrastructure.restapi.MetadataTestUtil.verifyMetadataErrorResponse;
+import static tech.pegasys.teku.spec.generator.signatures.NoOpLocalSigner.NO_OP_SIGNER;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.infrastructure.http.HttpErrorResponse;
+import tech.pegasys.teku.infrastructure.restapi.StubRestApiRequest;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.validator.api.GraffitiManager;
+import tech.pegasys.teku.validator.client.OwnedKeyManager;
+import tech.pegasys.teku.validator.client.Validator;
+
+class SetGraffitiTest {
+  private final DataStructureUtil dataStructureUtil =
+      new DataStructureUtil(TestSpecFactory.createDefault());
+  private final BLSPublicKey publicKey = dataStructureUtil.randomPublicKey();
+  private final String graffiti = "Test graffiti";
+
+  private final OwnedKeyManager keyManager = mock(OwnedKeyManager.class);
+  private final GraffitiManager graffitiManager = mock(GraffitiManager.class);
+  private final SetGraffiti handler = new SetGraffiti(keyManager, graffitiManager);
+  private final StubRestApiRequest request =
+      StubRestApiRequest.builder()
+          .metadata(handler.getMetadata())
+          .pathParameter("pubkey", publicKey.toHexString())
+          .build();
+
+  @Test
+  void shouldSuccessfullySetGraffiti() throws IOException {
+    request.setRequestBody(graffiti);
+
+    final Validator validator = new Validator(publicKey, NO_OP_SIGNER, Optional::empty);
+    when(keyManager.getValidatorByPublicKey(any())).thenReturn(Optional.of(validator));
+
+    handler.handleRequest(request);
+
+    assertThat(request.getResponseCode()).isEqualTo(SC_NO_CONTENT);
+    assertThat(request.getResponseBody()).isNull();
+  }
```

Comment:
Similar to the test on deleteGraffiti. We need to check that the POST API method is calling the graffitiManager.setGraffiti(..) method. Otherwise, we are not testing anything.